### PR TITLE
ZNTA-650 pages xml migration

### PR DESCRIPTION
**Pull request by huangp**
https://zanata.atlassian.net/browse/ZNTA-659

### CODE REVIEW

**Comment by seanf** [comment]
- [ ] I think someone with more JSF experience ought to review this (pages.xml migration) too.

File: zanata-war/src/main/webapp/WEB-INF/pages.xml [hunk 1]
```diff
@@ -172,62 +43,16 @@
     </navigation>
   </page>
 
-
   <!-- Admin -->
   <!-- TODO [CDI] find a way to secure a url pattern in CDI -->
   <page view-id="/admin/*" action="#{identity.checkRole('admin')}"></page>
```
Comment (carlosmunoz) [inline zanata-war/src/main/webapp/WEB-INF/pages.xml]: Worst case scenario we have to list out every single admin view in faces-config.xml

Comment (huangp) [inline zanata-war/src/main/webapp/WEB-INF/pages.xml]: We can't call action in faces-config.xml. It only controls page flow. The
action bit is done in every single xhtml. But yes worse case we may need to
do that. Or use another servlet filter

On Thu, Sep 24, 2015 at 2:54 PM, Carlos Munoz <notifications@github.com>
wrote:

> In zanata-war/src/main/webapp/WEB-INF/pages.xml
> <https://github.com/zanata/zanata-server/pull/959#discussion_r40285553>:
>
> > @@ -172,62 +43,16 @@
> >      </navigation>
> >    </page>
> >
> > -
> >    
> >    
> >    <page view-id="/admin/*" action="#{identity.checkRole('admin')}"></page>
>
> Worst case scenario we have to list out every single admin view in
> faces-config.xml
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/959/files#r40285553>.
>



-- 
Patrick Huang
Senior Software Engineer
Engineering - Internationalisation
Red Hat, Asia-Pacific Pty Ltd
Level 1, 193 North Quay
Brisbane 4000
Office: +61 7 3514 8278
Fax: +61 7 3514 8199
IRC: pahuang
github: github.com/huangp
Website: www.redhat.com

Comment (carlosmunoz) [inline zanata-war/src/main/webapp/WEB-INF/pages.xml]: I have no problem with adding the check to every admin xhtml page; in fact that's the approach we are taking with all other pages.
In fact I prefer that to a servlet filter as the it's clear what actions are being performed by looking at the view.

Comment (huangp) [inline zanata-war/src/main/webapp/WEB-INF/pages.xml]: Done in
https://github.com/zanata/zanata-server/commit/825e320af3cad8393031a6058a681b0e6736b944

On Thu, Sep 24, 2015 at 3:59 PM, Carlos Munoz <notifications@github.com>
wrote:

> In zanata-war/src/main/webapp/WEB-INF/pages.xml
> <https://github.com/zanata/zanata-server/pull/959#discussion_r40287936>:
>
> > @@ -172,62 +43,16 @@
> >      </navigation>
> >    </page>
> >
> > -
> >    
> >    
> >    <page view-id="/admin/*" action="#{identity.checkRole('admin')}"></page>
>
> I have no problem with adding the check to every admin xhtml page; in fact
> that's the approach we are taking with all other pages.
> In fact I prefer that to a servlet filter as the it's clear what actions
> are being performed by looking at the view.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/959/files#r40287936>.
>



-- 
Patrick Huang
Senior Software Engineer
Engineering - Internationalisation
Red Hat, Asia-Pacific Pty Ltd
Level 1, 193 North Quay
Brisbane 4000
Office: +61 7 3514 8278
Fax: +61 7 3514 8199
IRC: pahuang
github: github.com/huangp
Website: www.redhat.com

**Comment by huangp** [comment]
migrated to #981

**Pull request closed without merging**